File: src/main/java/com/github/stokito/IdeaSingletonInspection/quickFixes/InstanceGettersModifiersFix.java
package com.github.stokito.IdeaSingletonInspection.quickFixes;

import com.intellij.codeInspection.LocalQuickFix;
import com.intellij.codeInspection.ProblemDescriptor;
import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiMethod;
import org.jetbrains.annotations.NotNull;

public class InstanceGettersModifiersFix implements LocalQuickFix {

  @NotNull
  @Override
  public String getName() {
    return "Set public and static modifiers";
  }

  @Override
  public void applyFix(@NotNull final Project project, @NotNull final ProblemDescriptor descriptor) {
    @NotNull final PsiMethod instanceGetter = (PsiMethod)descriptor.getPsiElement();
    instanceGetter.getModifierList().setModifierProperty("public", true);
    instanceGetter.getModifierList().setModifierProperty("static", true);
  }

  @NotNull
  @Override
  public String getFamilyName() {
    return getName();
  }

}

File: src/main/java/com/github/stokito/IdeaSingletonInspection/quickFixes/InstanceGettersModifiersFix.kt
package com.github.stokito.IdeaSingletonInspection.quickFixes

import com.intellij.codeInspection.LocalQuickFix
import com.intellij.codeInspection.ProblemDescriptor
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiMethod
import com.intellij.psi.PsiModifier.PUBLIC
import com.intellij.psi.PsiModifier.STATIC

class InstanceGettersModifiersFix : LocalQuickFix {
    override fun getName(): String {
        return "Set public and static modifiers"
    }

    override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
        val instanceGetter = descriptor.psiElement as PsiMethod
        instanceGetter.modifierList.setModifierProperty(PUBLIC, true)
        instanceGetter.modifierList.setModifierProperty(STATIC, true)
    }

    override fun getFamilyName(): String {
        return name
    }
}
